clients/venice-producer/src/main/java/com/linkedin/venice/producer/VeniceProducer.java

```diff
@@ -0,0 +1,77 @@
+package com.linkedin.venice.producer;
+
+import com.linkedin.venice.writer.update.UpdateBuilder;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+
+/**
+ * The API for online applications to write to Venice. These APIs are only eventually consistent and the futures
+ * returned by them only guarantee that the write operation was durable. There is no option to block until the data
+ * is readable.
+ * @param <K> Key of the record that needs to be updated
+ * @param <V> Value that needs to be written
+ */
+public interface VeniceProducer<K, V> {
+  /**
+   * A write operation where a full value is written to replace the existing value.
+   * @param key Key of the record that needs to be updated
+   * @param value The full value that needs to be written
+   * @return A {@link CompletableFuture} that completes when the write operation is durable. It does not imply that the
+   *         data is available to readers.
+   */
+  CompletableFuture<DurableWrite> asyncPut(K key, V value);
```

Comment:
What's the A/A default/behavior when the user don't specify a logical time when writing to RT?